README.md

```diff
@@ -3,58 +3,136 @@
+Kubetest is a wrapper that includes everything needed to deploy a cluster, test it (using e2e.test), gather logs, then upload the results to a Google Storage account. It has built-in cloud provider scripts to build Linux+Windows clusters using Azure and GCP.
 
-### e2e.test
 
-Make sure you have a working [Kubernetes development environment](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md).
-```bash
-go get -d k8s.io/kubernetes
-cd $GOPATH/src/k8s.io/kubernetes
-./build/run.sh make WHAT=test/e2e/e2e.test
-```
-Once complete, the binary will be available at: `~/go/src/k8s.io/kubernetes/_output/dockerized/bin/linux/amd64/e2e.test`
+#### Azure
 
-If building on Mac or Windows, you can specify `KUBE_BUILD_PLATFORMS`.
+> TODO: This section is still under construction
 
-For Windows
-```bash
-./build/run.sh make KUBE_BUILD_PLATFORMS=windows/amd64
+Set environment variables:
+
+`AZURE_SSH_PUBLIC_KEY_FILE` - Path to the SSH public key you want to use for connecting to the cluster nodes. This is probably `~/.ssh/id_rsa.pub`
+
+`AZURE_CREDENTIALS` - Path to a TOML file with a service account credential that will be used for creating the Azure resources
+
+```toml
+[Creds]
+  ClientID = ""
+  ClientSecret = ""
+  SubscriptionId = ""
+  TenantID = ""
+  StorageAccountName = ""
+  StorageAccountKey = ""
 ```
 
-For Mac
+Once those are set, you can run `kubetest` and it will do the rest. The full set of tests will take 6-7 hours.
+
 ```bash
-./build/run.sh make KUBE_BUILD_PLATFORMS=darwin/amd64
+export KUBECONFIG=path/to/kubeconfig
+curl https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/images/image-repo-list-ws2019 -o repo_list
+export KUBE_TEST_REPO_LIST=$(pwd)/repo_list
+export AZURE_CREDENTIALS=TODO
+kubetest --test=true \
+  --up=true \
+  --down=true \
+  --deployment=acsengine \
+  --provider=skeleton \
+  --build=bazel \
+  --acsengine-location=westus \
+  --acsengine-admin-username=azureuser \
+  --acsengine-admin-password=MakeItSecure123! \
+  --acsengine-creds=$AZURE_CREDENTIALS \
+  --acsengine-download-url=https://github.com/Azure/aks-engine/releases/download/v0.30.0/aks-engine-v0.30.0-linux-amd64.tar.gz \
+  --acsengine-public-key=$AZURE_SSH_PUBLIC_KEY_FILE \
+  --acsengine-winZipBuildScript=https://raw.githubusercontent.com/Azure/acs-engine/master/scripts/build-windows-k8s.sh \
+  --acsengine-orchestratorRelease=1.13 \
+  --acsengine-template-url=https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/job-templates/kubernetes_release.json \
+  --acsengine-agentpoolcount=3 \
+  --test_args=--node-os-distro=windows --ginkgo.focus=\\[Conformance\\]|\\[NodeConformance\\]|\\[sig-windows\\]|\\[sig-apps\\].CronJob --ginkgo.skip=\\[LinuxOnly\\]|\\[k8s.io\\].Pods.*should.cap.back-off.at.MaxContainerBackOff.\\[Slow\\]\\[NodeConformance\\]|\\[k8s.io\\].Pods.*should.have.their.auto-restart.back-off.timer.reset.on.image.update.\\[Slow\\]\\[NodeConformance\\] \
+  --timeout=450m
```

Comment:
I wouldn't add the timeout. In a non automated scenario there isn't really any need for this. It just kills the ginkgo process and it might prevent results to be properly logged ( if you use it with --dump for example ).